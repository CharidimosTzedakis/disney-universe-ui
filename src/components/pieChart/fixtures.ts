export const pieChartDataPoints = [
  {
    name: "Mikey",
    y: 2,
    films: ["Terminator", "Back to the future"],
  },
  {
    name: "Donald",
    y: 4,
    films: ["Blade runner", "film2", "film3", "film4"],
  },
  {
    name: "Taz",
    y: 7,
    films: ["film5", "film6", "film7", "film8", "film9", "film10", "film11"],
  },
  { name: "char1", y: 0, films: [] },
];
