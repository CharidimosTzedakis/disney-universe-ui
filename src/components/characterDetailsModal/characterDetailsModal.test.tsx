import { expect, describe, it, vi } from "vitest";
import { render } from "@testing-library/react";
import { Provider } from "urql";
import { never, fromValue } from "wonka";
import CharacterDetailsModal from "./characterDetailsModal";
import { charactersWithDetails } from "@test/fixtures";

vi.mock(import("urql"), async (importOriginal) => {
  const actual = await importOriginal();
  return {
    ...actual,
    seQuery: vi.fn().mockReturnValue([{ fetching: true }]),
  };
});

describe("CharacterDetailsModal", () => {
  const fetchingState = {
    executeQuery: () => never,
  };

  it("renders correctly in loading state", async () => {
    render(
      <Provider value={fetchingState}>
        <CharacterDetailsModal
          isOpen={true}
          selectedCharacter={{ id: 2, name: "Achilles" }}
          onClose={() => {}}
        />
      </Provider>,
    );
    expect(document.body).toMatchSnapshot();
  });

  it("renders details for a specific character", async () => {
    const fetchingState = {
      executeQuery: () =>
        fromValue({
          data: {
            characters: {
              items: [charactersWithDetails[0]],
            },
          },
        }),
    };

    render(
      <Provider value={fetchingState}>
        <CharacterDetailsModal
          isOpen={true}
          selectedCharacter={{ id: 1, name: "Achilles" }}
          onClose={() => {}}
        />
      </Provider>,
    );
    expect(document.body).toMatchSnapshot();
  });
});
