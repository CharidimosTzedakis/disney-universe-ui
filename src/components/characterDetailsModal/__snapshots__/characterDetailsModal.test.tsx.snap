// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CharacterDetailsModal > renders correctly in loading state 1`] = `
<body>
  <div />
  <div>
    <div
      class="ant-modal-root css-dev-only-do-not-override-142vneq"
    >
      <div
        class="ant-modal-mask"
      />
      <div
        class="ant-modal-wrap"
        tabindex="-1"
      >
        <div
          aria-labelledby="test-id"
          aria-modal="true"
          class="ant-modal css-dev-only-do-not-override-142vneq ant-zoom-appear ant-zoom-appear-prepare ant-zoom"
          role="dialog"
          style="width: 520px;"
        >
          <div
            style="outline: none;"
            tabindex="0"
          >
            <div
              class="ant-modal-content"
            >
              <button
                aria-label="Close"
                class="ant-modal-close"
                type="button"
              >
                <span
                  class="ant-modal-close-x"
                >
                  <span
                    aria-label="close"
                    class="anticon anticon-close ant-modal-close-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close"
                      fill="currentColor"
                      fill-rule="evenodd"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
              <div
                class="ant-modal-header"
              >
                <div
                  class="ant-modal-title"
                  id="test-id"
                >
                  Achilles
                </div>
              </div>
              <div
                class="ant-modal-body"
              >
                <div
                  class="ant-skeleton ant-skeleton-active ant-modal-body-skeleton css-dev-only-do-not-override-142vneq"
                >
                  <div
                    class="ant-skeleton-content"
                  >
                    <ul
                      class="ant-skeleton-paragraph"
                    >
                      <li />
                      <li />
                      <li />
                      <li
                        style="width: 61%;"
                      />
                    </ul>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`CharacterDetailsModal > renders details for a specific character 1`] = `
<body>
  <div />
  <div>
    <div
      class="ant-modal-root css-dev-only-do-not-override-142vneq"
    >
      <div
        class="ant-modal-mask"
      />
      <div
        class="ant-modal-wrap"
        tabindex="-1"
      >
        <div
          aria-labelledby="test-id"
          aria-modal="true"
          class="ant-modal css-dev-only-do-not-override-142vneq ant-zoom-appear ant-zoom-appear-prepare ant-zoom"
          role="dialog"
          style="width: 520px;"
        >
          <div
            style="outline: none;"
            tabindex="0"
          >
            <div
              class="ant-modal-content"
            >
              <button
                aria-label="Close"
                class="ant-modal-close"
                type="button"
              >
                <span
                  class="ant-modal-close-x"
                >
                  <span
                    aria-label="close"
                    class="anticon anticon-close ant-modal-close-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close"
                      fill="currentColor"
                      fill-rule="evenodd"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
              <div
                class="ant-modal-header"
              >
                <div
                  class="ant-modal-title"
                  id="test-id"
                >
                  Achilles
                </div>
              </div>
              <div
                class="ant-modal-body"
              >
                <div
                  class="ant-image css-dev-only-do-not-override-142vneq"
                  style="width: 400px;"
                >
                  <img
                    alt="Achilles"
                    class="ant-image-img css-dev-only-do-not-override-142vneq"
                    src="src="https://static.wikia.nocookie.net/disney/images/d/d3/Vlcsnap-2015-05-06-23h04m15s601.png""
                    width="400"
                  />
                </div>
                <h5
                  class="ant-typography css-dev-only-do-not-override-142vneq"
                >
                  TV shows
                </h5>
                <ul>
                  <li>
                    Hercules (TV series)
                  </li>
                </ul>
                <h5
                  class="ant-typography css-dev-only-do-not-override-142vneq"
                >
                  Video games
                </h5>
                <ul>
                  <li>
                    Kingdom Hearts III
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
